workflow-ui/compose/src/main/java/com/squareup/workflow1/ui/compose/ComposedViewFactory.kt

```diff
@@ -0,0 +1,141 @@
+// See https://youtrack.jetbrains.com/issue/KT-31734
+@file:Suppress("RemoveEmptyParenthesesFromAnnotationEntry")
+
+package com.squareup.workflow1.ui.compose
+
+import android.content.Context
+import android.view.View
+import android.view.ViewGroup
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.platform.ComposeView
+import com.squareup.workflow1.ui.LayoutRunner
+import com.squareup.workflow1.ui.ViewEnvironment
+import com.squareup.workflow1.ui.ViewFactory
+import com.squareup.workflow1.ui.WorkflowUiExperimentalApi
+import com.squareup.workflow1.ui.bindShowRendering
+import kotlin.reflect.KClass
+
+/**
+ * Creates a [ViewFactory] that uses a [Composable] function to display the rendering.
+ *
+ * Simple usage:
+ *
+ * ```
+ * val FooViewFactory = composedViewFactory { rendering, _ ->
+ *   Text(rendering.message)
+ * }
+ *
+ * …
+ *
+ * val viewRegistry = ViewRegistry(FooViewFactory, …)
+ * ```
+ *
+ * If you need to write a class instead of a function, for example to support dependency injection,
+ * see [ComposedViewFactory].
+ *
+ * For more details about how to write composable view factories, see [ComposedViewFactory].
+ */
+@WorkflowUiExperimentalApi
+public inline fun <reified RenderingT : Any> composedViewFactory(
```

Comment:
What's the thinking behind `composedViewFactory` rather than `composeViewFactory`? 

(I get that the latter could be a hang-up in that the uninitiated might read 'compose' as a verb. But – being initiated on the Compose side of things – I found myself wondering whether the past participle wasn't mean to convey some subtlety of its own. Maybe there's no winning. A data point for your consideration, at any rate.)

Reply:
i don't care what we call it

Reply:
No big opinions here either, but my mind wanted `composeViewFactory`, seemingly. I think Oleg may have asked the same question about the name in your retro design doc.

Reply:
I did a quick poll on both [workflow](https://workflow-community.slack.com/archives/C013JGW58N7/p1629240387012500) and [internal](https://square.slack.com/archives/CE470RL5T/p1629240405170000) slack, and `composeViewFactory` won in both. I'll change it.